cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,169 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type lintUtils interface {
+	Glob(pattern string) (matches []string, err error)
+
+	getExecRunner() execRunner
+	getGeneralPurposeConfig(configURL string) error
+}
+
+type lintUtilsBundle struct {
+	*piperutils.Files
+	execRunner *command.Command
+	client     *piperhttp.Client
+}
+
+func newLintUtilsBundle() *lintUtilsBundle {
+	return &lintUtilsBundle{
+		Files:  &piperutils.Files{},
+		client: &piperhttp.Client{},
+	}
+}
+
+func (u *lintUtilsBundle) getExecRunner() execRunner {
+	if u.execRunner == nil {
+		u.execRunner = &command.Command{}
+		u.execRunner.Stdout(log.Writer())
+		u.execRunner.Stderr(log.Writer())
+	}
+	return u.execRunner
+}
+
+func (u *lintUtilsBundle) getGeneralPurposeConfig(configURL string) error {
+	response, err := u.client.SendRequest(http.MethodGet, configURL, nil, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("error reading %v: %w", response.Body, err)
+	}
+
+	err = u.FileWrite(filepath.Join(".pipeline", ".eslintrc.json"), content, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to write .eslintrc.json file to .pipeline/: %w", err)
+	}
+
+	return nil
+}
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := newLintUtilsBundle()
+	npmExecutorOptions := npm.ExecutorOptions{RunScripts: []string{"ci-lint"}, RunOptions: []string{"--silent"}, DefaultNpmRegistry: config.DefaultNpmRegistry, SapNpmRegistry: config.SapNpmRegistry, ExecRunner: utils.getExecRunner()}
+	npmExecutor, err := npm.NewExecutor(npmExecutorOptions)
+
+	err = runNpmExecuteLint(npmExecutor, utils, &config)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(npmExecutor npm.Executor, utils lintUtils, config *npmExecuteLintOptions) error {
+	packageJSONFiles := npmExecutor.FindPackageJSONFiles()
+	packagesWithCiLint, _ := npmExecutor.FindPackageJSONFilesWithScript(packageJSONFiles, "ci-lint")
+
+	if len(packagesWithCiLint) > 0 {
+		err := runCiLint(npmExecutor, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := runDefaultLint(npmExecutor, utils, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runCiLint(npmExecutor npm.Executor, failOnError bool) error {
+	err := npmExecutor.ExecuteAllScripts()
```

Comment:
Also from the API I guess I would have prefert something where the scripts which are executed are part of the parameters, i.e. RunScripts could be a parameter and not part of state of the executor which is defined quite far away.

Also it felt somewhat not intuitive that you have to pass runScripts where it is not required, e.g. in the mta step: https://github.com/SAP/jenkins-library/pull/1636/files#diff-14cae8ca63f373c3486ee76e2e671c40R85